### ISSUE TITLE
Screen flashes white when loading content in new tab or reopening Brave in Dark/Night Mode

### ISSUE THREAD

**Issue by Saoiray** [brave/brave-browser#42929]
### Description

While Brave is set to Night Mode and appearance to Dark, a white "flash" appears when loading content in a tab for the first time or reopening the browser.

Once the flash occurs, it behaves as expected for the rest of the session. However, it happens again if you open a new tab or exit and relaunch the browser.

### Steps to reproduce

Scenario 1: Opening a new tab

1.  Set Brave to Dark Mode and enable Night Mode (on iOS).
2. Open a new tab.
3. Navigate to a website.

Scenario 2: Reopening Brave

1. Ensure tabs are already open, with Dark/Night mode enabled.
2. Dismiss Brave (swipe away from app switcher).
3. Reopen Brave.
4. The first tab loaded will flash white.

### Actual result

The screen temporarily flashes bright white before loading content.

![Image](https://github.com/user-attachments/assets/8a3b0c13-8faf-41eb-98b2-961f6314e54b)

### Expected result

There should be no bright white flash while in Dark Mode.

### Reproduces how often

Easily reproduced

### Brave version

1.73 (97)

### Device/iOS version

* iPhone 13 Pro Max
* iOS 18.1.1
***

Also reported by someone with:
* iPhone 14 Pro 
* 18.1.1

### Affected browser versions

- [ ] latest AppStore
- [ ] latest TestFlight
- [ ] previous TestFlight

### Reproducibility

- [x] with Brave Shields disabled
- [ ] in the latest version of mobile Safari

### Miscellaneous information

Reported from https://community.brave.com/t/every-page-flashes-white-while-loading/586852

**Comment by Saoiray** [comment]
Had seen there was an iOS update. Just updated to iOS 18.2 but no change to the issue.

**Comment by bastele7** [comment]
I can also confirm that annoying behavior on Brave for Android. I use Brave 1.73.101.

**Comment by wdiable** [comment]
Commenting to receive updates

**Comment by Uni-verse** [comment]
Verified on `iPhone XR` running `iOS 17.7` using version `1.75 (164)`

* Ensured that screen does not flash white when running through both scenarios in https://github.com/brave/brave-browser/issues/42929#issue-2743743871

example | example
---|---
![Image](https://github.com/user-attachments/assets/a9764b55-ddad-4635-9c7a-75824ac082c7) | ![Image](https://github.com/user-attachments/assets/5adbdc70-d011-4c30-9231-17f039457624)

**Comment by Uni-verse** [comment]
Verified on `iPad (9th Gen)` running `iPadOS 18.2.1` running `1.75 (164)`

* Ensured that screen does not flash white when running through both scenarios in https://github.com/brave/brave-browser/issues/42929#issue-2743743871

example | example
---|---
![Image](https://github.com/user-attachments/assets/5315c2c8-c35a-40d7-bfe4-31cd9a2663ff) | ![Image](https://github.com/user-attachments/assets/6b808377-e0b9-48e0-92c0-bfec7c4e8c81)

**Comment by arshad4000** [comment]
https://github.com/user-attachments/assets/f896caf8-7c5c-47c4-ac5c-543f683a9873


Was following this issue. Noticed that it isn't fixed in the latest beta version. Attaching a screen recording of the behavior.

Brave version : Version 1.75 (166) BraveCore 1.75.166 (132.0.6834.111) Device iPhone16,1 (iOS 18.2.1)

**Comment by kylehickinson** [comment]
@arshad4000 the fix is only applies when you have night mode enabled due to webcompat issues

**Comment by arshad4000** [comment]
> @arshad4000 the fix is only applies when you have night mode enabled due to webcompat issues 

@kylehickinson Yes, I am not able to reproduce with the night mode turned on. I am not sure if there is a way to fix this issue without affecting webcompat - because on safari and some other browsers this issue doesn't exist.

**Comment by wdiable** [comment]
Working perfectly for me in the beta build 🙏🏻

**Comment by wdiable** [comment]
I've noticed in the latest beta that when you go back a page using the swipe action (left of screen to right), the background is still while.

Example:

![Image](https://github.com/user-attachments/assets/c0c7bd9b-00fd-49ac-b81c-f1a4116aa291)

Example using Safari with dark reader extension:

![Image](https://github.com/user-attachments/assets/3f6e6e64-0540-416c-a527-75416368ebee)
